packages/nns/src/token.ts

```diff
@@ -119,14 +119,14 @@ export class TokenAmount {
    */
   public static fromNumber({
     amount,
-    token = ICPToken,
+    token,
   }: {
     amount: number;
-    token?: Token;
-  }): TokenAmount | FromStringToTokenError {
-    // If more than TOKEN_DECIMAL_ACCURACY, it returns 0, not the error.
-    return TokenAmount.fromString({
-      amount: amount.toFixed(TOKEN_DECIMAL_ACCURACY),
+    token: Token;
+  }): TokenAmount {
+    const e8s = BigInt(Math.floor(amount * Number(E8S_PER_TOKEN)));
```

Comment:
`TOKEN_DECIMAL_ACCURACY` is probably not used anymore and can be deleted from the ` constant`?

Reply:
oops, sorry, my bad! I'll do that in another PR.

Reply:
https://github.com/dfinity/ic-js/pull/206